lib/Docs/RST/RSTCopier.php

```diff
@@ -157,11 +159,18 @@ private function getSidebarRST(string $docsDir): string
             $sidebar = $this->rstFileRepository->getFileContents($sidebarPath);
 
             // sidebar.rst paths were wrong. Remove this once sidebar.rst is updated everywhere
-            $sidebar = preg_replace('/^([ \t]*)(reference|tutorials|cookbook)/m', '$1/$2', $sidebar);
-
-            return $sidebar;
+            return $this->pregReplace('/^([ \t]*)(reference|tutorials|cookbook)/m', '$1/$2', $sidebar);
         }
 
         return self::DEFAULT_SIDEBAR;
     }
+
+    private function pregReplace(string $pattern, string $replacement, string $content): string
```

Comment:
It's weird to create a method to make phpstan happy.

Reply:
It's worth a try.

Reply:
I used the Safe PHP library to test the preg_replace. We trade the ` Parameter ... of function preg_replace expects array|string, string|null given.` error with `should return string but returns array<string>|string.`, but not in the same amount of errors as before.

Reply:
On the one hand, it doesn't look to bad, and I think the new issue could be fixed by contributing generics if really it's an issue, or by wrapping that function in our own function that would only accept strings.

On the other hand, the last merged MR on safe was merged on April 13th… so I'm not sure how, well, _safe_ it is to rely on that library. I'd like to know what other maintainers think.

Reply:
We could either create our own safe preg_replace or create a class with the functionality to return a string. I doubt that we will be going to level 9 without multiple ignored-errors, so I don't see that we'll need more safe functions in the coming weeks. And it would be one less lib in the dependencies.

The pattern in preg_replace can be an array too, so we would be able to reduce the usage of that function and therefore less asserts and I wonder if the order of method invocations, together with str_replace is relevant. Maybe we can reduce the usage of preg_replace in this class.